### PR TITLE
Add ocp-4.4 provider

### PR DESCRIPTION
**Pull request by qinqon**
Also it uses OVNKubernetes as this is going to be the de-facto SDN implementation.

### CODE REVIEW

**Comment by dankenigsberg** [comment]
It is GREAT to see work on OCP-4.4, but I think that it is time that we have an OCP provider with `networking.networkType=OVNKubernetes`.

https://docs.openshift.com/container-platform/4.3/installing/installing_aws/installing-aws-network-customizations.html#nw-operator-configuration-parameters-for-ovn-sdn_installing-aws-network-customizations

File: cluster-provision/okd/scripts/provision.sh [hunk 1]
```diff
@@ -122,6 +122,10 @@ envsubst < /manifests/okd/registries.yaml > /registries.yaml
 set +x
 export PULL_SECRET=$(cat /etc/installer/token)
 export SSH_PUBLIC_KEY="ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkTkyrtvp9eWW6A8YVr+kz4TjGYe7gHzIw+niNltGEFHzD8+v1I2YJ6oXevct1YeS0o9HZyN1Q9qgCgzUFtdOKLv6IedplqoPkcmF0aYet2PkEDo3MlTBckFXPITAMzF8dJSIFo9D8HfdOV0IAdx4O7PtixWKn5y2hMNG0zQPyUecp4pzC6kivAIhyfHilFR61RGL+GPXQ2MWZWFYbAGjyiYJnAmCP3NOTd0jMZEnDkbUvxhMmBYSdETk1rRgm+R4LOzFUGaHqHDLKLX+FIPKcF96hrucXzcWyLbIbEgE98OHlnVYCzRdK8jlqm8tehUc9c9WhQ== vagrant insecure public key"
+
+# For ocp-4.4 we want OVNKubernetes
```
Comment (phoracek) [inline cluster-provision/okd/scripts/provision.sh]: I don't see how is this comment relevant or connected to the next line. How would someone use OVN networking? Do we need another provider for that?

File: cluster-up/cluster/ocp-4.3/provider.sh [hunk 2]
```diff
@@ -51,6 +51,9 @@ function up() {
         params=" --container-registry= $params"
     fi
 
+    # Free some cached/buffered mem
+    sync; echo 3 > /proc/sys/vm/drop_caches
```
Comment (phoracek) [inline cluster-up/cluster/ocp-4.3/provider.sh]: This is not related to OCP-4.4. If you need to do a fix here, please elaborate in the commit message.

File: cluster-up/cluster/ocp-4.4/README.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
+# OCP 4.4 in ephemeral containers
```
Comment (phoracek) [inline cluster-up/cluster/ocp-4.4/README.md]: This is the place to describe the network plugin that is used

File: cluster-up/cluster/ocp-4.4/provider.sh [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+    # If provision test mode is on, use local image
+    if [ -z $KUBEVIRTCI_PROVISION_CHECK ]; then
+        params=" --container-registry ${container_registry} $params"
+    else
+        params=" --container-registry= $params"
+    fi
+
+    # Free some cached/buffered mem
+    sync; echo 3 > /proc/sys/vm/drop_caches
```
Comment (phoracek) [inline cluster-up/cluster/ocp-4.4/provider.sh]: If we have it here, do we need it on the other places?

File: cluster-provision/okd/scripts/provision.sh (hunk 1 above)
Comment (qinqon) [reply cluster-provision/okd/scripts/provision.sh]: This is old stuff, let's me move it to the proper place.

Comment (qinqon) [reply cluster-provision/okd/scripts/provision.sh]: Ohh gosh I have "export NETWORKING_TYPE=OpenShiftSDN" harcoded at provision.sh

File: cluster-up/cluster/ocp-4.4/provider.sh (hunk 4 above)
Comment (qinqon) [reply cluster-up/cluster/ocp-4.4/provider.sh]: It's done at ocp-4.3 too

Comment (phoracek) [reply cluster-up/cluster/ocp-4.4/provider.sh]: It is done in ocp-4.3, here and in provision.sh. Do we need it to be on all 3?

Comment (phoracek) [reply cluster-up/cluster/ocp-4.4/provider.sh]: @qinqon

File: cluster-provision/okd/scripts/provision.sh [hunk 5]
```diff
@@ -122,10 +122,15 @@ envsubst < /manifests/okd/registries.yaml > /registries.yaml
 set +x
 export PULL_SECRET=$(cat /etc/installer/token)
 export SSH_PUBLIC_KEY="ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkTkyrtvp9eWW6A8YVr+kz4TjGYe7gHzIw+niNltGEFHzD8+v1I2YJ6oXevct1YeS0o9HZyN1Q9qgCgzUFtdOKLv6IedplqoPkcmF0aYet2PkEDo3MlTBckFXPITAMzF8dJSIFo9D8HfdOV0IAdx4O7PtixWKn5y2hMNG0zQPyUecp4pzC6kivAIhyfHilFR61RGL+GPXQ2MWZWFYbAGjyiYJnAmCP3NOTd0jMZEnDkbUvxhMmBYSdETk1rRgm+R4LOzFUGaHqHDLKLX+FIPKcF96hrucXzcWyLbIbEgE98OHlnVYCzRdK8jlqm8tehUc9c9WhQ== vagrant insecure public key"
+
 envsubst < /manifests/okd/install-config.yaml > ${INSTALL_CONFIG_FILE}
 unset PULL_SECRET
 set -x
 
+cat ${INSTALL_CONFIG_FILE}
+exit 1
```
Comment (phoracek) [inline cluster-provision/okd/scripts/provision.sh]: Leftover?

Comment (qinqon) [reply cluster-provision/okd/scripts/provision.sh]: Yep, totally, removed.

File: cluster-up/cluster/images.sh [hunk 6]
```diff
@@ -17,6 +17,7 @@ if [ -z $KUBEVIRTCI_PROVISION_CHECK ]; then
     IMAGES[okd-4.2]="okd-4.2@sha256:a830064ca7bf5c5c2f15df180f816534e669a9a038fef4919116d61eb33e84c5"
     IMAGES[okd-4.3]="okd-4.3@sha256:63abc3884002a615712dfac5f42785be864ea62006892bf8a086ccdbca8b3d38"
     IMAGES[ocp-4.3]="ocp-4.3@sha256:03a8c736263493961f198b5cb214d9b1fc265ece233c60bdb1c8b8b4b779ee1e"
+    IMAGES[ocp-4.4]="ocp-4.4@sha256:b235e87323ed88c46fedf27e9115573b92f228a82559ab7523dd1be183f66af8"
```
Comment (dankenigsberg) [inline cluster-up/cluster/images.sh]: how can we verify that the image we have built is based on the slightly-more stable images pointed from https://mirror.openshift.com/pub/openshift-v4/clients/ocp-dev-preview/ ?

Random images typically die after a few days. Linked images would survive months. Or so I'm told.
